static/src/javascripts/projects/commercial/modules/dfp/prepare-a9.js

```diff
@@ -10,28 +10,28 @@ import { shouldIncludeOnlyA9 } from 'commercial/modules/header-bidding/utils';
 import { amazonA9Test } from 'common/modules/experiments/tests/amazon-a9';
 import { isInVariantSynchronous } from 'common/modules/experiments/ab';
 
-let moduleLoadResult = Promise.resolve();
-if (!isGoogleProxy()) {
-    moduleLoadResult = import('lib/a9-apstag.js');
-}
-
-const setupA9: () => Promise<void> = () =>
-    moduleLoadResult.then(() => {
-        if (
-            shouldIncludeOnlyA9 ||
-            (dfpEnv.hbImpl.a9 &&
-                isInVariantSynchronous(amazonA9Test, 'variant') &&
-                commercialFeatures.dfpAdvertising &&
-                !commercialFeatures.adFree &&
-                !config.get('page.hasPageSkin') &&
-                !isGoogleProxy())
-        ) {
+const setupA9: () => Promise<void> = () => {
+    let moduleLoadResult = Promise.resolve();
```

Comment:
This is a little jarring to read with the let and mutation. Would it be simpler to written as an if / else?

```js
if (...) {
  import('lib/a9-apstag.js').then(() => {
    a9.initialise();
  }
} else {
  return Promise.resolve();
}
```

Reply:
I had a typo in my suggestion (fixed) which made it look simpler than it was. I've corrected it now and it's not as much of a win anymore so as I said, up to you.

Reply:
I tried that before settling on the current solution but I got a type mismatch error (Cannot assign function to `setupA9` because undefined [1] is incompatible with `Promise` [2] in the return value.), forcing me to do the second return in your example outside the else to appease the Flow. However if I do that I won't return the correct Promise when A9 runs.